File: local-plugins/fastspell.nvim/jslib/src/types/responses.ts
interface SpellingProblem{
    lineStart: number,
    lineOfset: number,
    word: string
    
}

interface SpellCheckResponse{
    kind: "lint"
    problems: Array<SpellingProblem>
}

interface SuggestionResponse {
    kind: "suggestion",
    suggestion: Array<string>
}

interface ErrorResponse {
    kind: "error",
    message: string
}

interface ConfigureSpellCheckResponse {
    kind: "configure_spell_check_response"
}

type SpellResponse = SpellCheckResponse | ErrorResponse | SuggestionResponse | ConfigureSpellCheckResponse;

export {SpellResponse, SpellCheckResponse, SuggestionResponse, ConfigureSpellCheckResponse}

File: local-plugins/fastspell.nvim/lua/types/responses.lua
---@class SpellingProblem
---@field lineStart number
---@field lineOfset number
---@field word string

---@class SpellCheckResponse
---@field kind "lint" # Literal string type
---@field problems SpellingProblem[] # Array of SpellingProblem

---@class LintError
---@field kind "error" # Literal string type
---@field message string

---@class ConfigureSpellCheckResponse
---@field kind "configure_spell_check_response"

---@alias SpellResponse SpellCheckResponse|LintError|ConfigureSpellCheckResponse
